include/eve/pattern.hpp

```diff
@@ -0,0 +1,138 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2018-2021 Joel FALCOU
+  Copyright 2018-2021 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <utility>
+#include <ostream>
+#include <eve/concept/pattern.hpp>
+
+namespace eve
+{
+  //================================================================================================
+  // Index-based pattern holder
+  //================================================================================================
+  template<std::ptrdiff_t... I> struct pattern_t
+  {
+    static inline constexpr auto size(int) noexcept { return sizeof...(I); }
```

Comment:
Why does this have an int parameter?

Reply:
I think it is now redundant. it was used in the old version but it needs removal